src/Avalonia.Base/Data/InstancedBinding.cs

```diff
@@ -106,7 +107,7 @@ internal InstancedBinding(object? value, BindingMode mode, BindingPriority prior
         {
             _ = observer ?? throw new ArgumentNullException(nameof(observer));
 
-            return new InstancedBinding(observer, BindingMode.OneWayToSource, priority);
+            return new InstancedBinding((IObservable<object?>)observer, BindingMode.OneWayToSource, priority);
```

Comment:
Is it always a valid operation?
I can imagine having InstancedBinding with only IObserver implementation and without IObservable. And I suppose our binding system actually doesn't require IObservable in other places.

Reply:
I wish it was possible to define API like:
```
public (IObservable | IObserver) Source { get; }
```
But that's not a TypeScript for better or worse.

Reply:
Yeah I did have that concern, but previously it used an `ISubject` so just an `IObserver` wouldn't have been possible anyway. Not too fond of this hack, but given that we're mainly keeping `InstancedBinding`  public for API compatibility it doesn't worry me that much,